zerver/actions/message_send.py

```diff
@@ -913,6 +915,41 @@ def do_send_messages(
             # assert needed because stubs for django are missing
             assert send_request.stream is not None
             realm_id = send_request.stream.realm_id
+            sender = send_request.message.sender
+
+            # The sender automatically follows the topic if the sender has
+            # 'automatically_follow_topics_policy' set to 'AUTOMATICALLY_FOLLOW_TOPICS_ON_PARTICIPATION'.
+            if (
+                not sender.is_bot
+                and sender.automatically_follow_topics_policy
+                == UserProfile.AUTOMATICALLY_FOLLOW_TOPICS_ON_PARTICIPATION
+            ):
+                do_set_user_topic_visibility_policy(
+                    user_profile=sender,
+                    stream=send_request.stream,
+                    topic=send_request.message.topic_name(),
+                    visibility_policy=UserTopic.VisibilityPolicy.FOLLOWED,
+                )
+
+            # The sender automatically follows the topic if the sender has started the topic and
+            # 'automatically_follow_topics_policy' set to 'AUTOMATICALLY_FOLLOW_TOPICS_ON_INITIATION'.
+            accessible_messages_in_topic = UserMessage.objects.filter(
+                user_profile=sender,
+                message__recipient_id=send_request.message.recipient_id,
+                message__subject__iexact=send_request.message.topic_name(),
+            )
```

Comment:
We can avoid the query if the policy isn't `ON_PARTICIPATION`, right? Currently we are running this query for all messages.
An `if-elif` sequence would be better here:
```
if not bot:
    if policy = ON_PARTICIPATION:
        # ...
    elif policy = ON_INITIATION:
        accessible_messages_in_topic = ...
        if len(accessible_messages_in_topic) > 0:
            # ...
```